### PR TITLE
RN71 upgrade

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
RN71 upgrade

## Changelog
RN71 upgrade

### CODE REVIEW

**Comment by Inbal-Tish** [comment]
@ethanshar I disabled the Incubator.Dialog test for now. I'll keep looking for a solution for it but I don't think it should block this upgrade

File: src/commons/Constants.ts [hunk 1]
```diff
@@ -131,15 +131,14 @@ const constants = {
       (screenHeight >= 812 || screenWidth >= 812);
   },
   /* Orientation */
+  dimensionsEventListener: undefined,
```
Comment (ethanshar) [inline src/commons/Constants.ts:134]: Is `dimensionsEventListener` being used anywhere?

File: src/components/view/index.tsx [hunk 2]
```diff
@@ -142,6 +142,7 @@ function View(props: ViewProps, ref: any) {
 
   return (
     <ViewContainer
+      //@ts-ignore
```
Comment (ethanshar) [inline src/components/view/index.tsx:145]: Is it something that might be fixed by react-native types?
Maybe use `@ts-expect-error` instead?

**Review by ethanshar — COMMENTED** [review]
Looks good. 
Wrote some comments. 

I do have an issue on IOS (Android is fine) in the main screen when scrolling down the list of screens it throws an error, something related to the sections list. 
Do you get it?

**Comment by ethanshar** [comment]
@Inbal-Tish One more thing. 
It's worth someone with the new Mac to checkout this PR and see the new version works fine. either @M-i-k-e-l  or @adimorwix  can check it, they have the new mac

**Comment by Inbal-Tish** [comment]
> Looks good. Wrote some comments.
> 
> I do have an issue on IOS (Android is fine) in the main screen when scrolling down the list of screens it throws an error, something related to the sections list. Do you get it?

Fixed!

File: src/incubator/Dialog/__tests__/index.spec.tsx [hunk 3]
```diff
@@ -49,13 +49,13 @@ describe('Incubator.Dialog', () => {
     const component = _TestCase({onDismiss});
     const dialogDriver = new ComponentDriver({component, testID: 'dialog'});
     expect(await dialogDriver.exists()).toBeFalsy();
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/incubator/Dialog/__tests__/index.spec.tsx:51]: Ticket #3447 created

File: src/components/view/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/components/view/index.tsx]: Changed. Don't really know where this error comes from, might be a RN bug...

File: src/commons/Constants.ts (hunk 1 above)
Comment (Inbal-Tish) [reply src/commons/Constants.ts:134]: Yes, It is used for fixing the layout when orientation change in Carousel, ColorPalette, GridView, Slider, Wizard, and useOrientation hook.

File: demo/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply demo/tsconfig.json:1]: Removed. I wasn't sure about this one and forgot to recheck after I finished with fixing the builds... Thanks!

File: .eslintrc.js [hunk 5]
```diff
@@ -43,6 +43,7 @@ module.exports = {
     '@typescript-eslint/no-empty-function': 0,
     '@typescript-eslint/camelcase': 0,
     '@typescript-eslint/indent': 0,
-    '@typescript-eslint/explicit-module-boundary-types': 0
+    '@typescript-eslint/explicit-module-boundary-types': 0,
+    semi: [0, 'never']
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply .eslintrc.js:47]: It was hiding 27 errors so I though we were not enforcing it and now it's started erroring when I updated the library. I can fix these errors and remove this

**Comment by Inbal-Tish** [comment]
> @Inbal-Tish One more thing. It's worth someone with the new Mac to checkout this PR and see the new version works fine. either @M-i-k-e-l or @adimorwix can check it, they have the new mac

Ok. I'll ask them to check it.

**Comment by M-i-k-e-l** [comment]
Both platforms installed and small sanity was performed

**Review by ethanshar — APPROVED** [review]

**Pull request merged**